cmd/client-keystone-auth/main.go

```diff
@@ -70,59 +70,92 @@ func promptForString(field string, r io.Reader, show bool) (result string, err e
 			return "", fmt.Errorf("error reading input for %s", field)
 		}
 	}
+
+	// intentionally ignore "unexpected newline" errors to allow read empty lines from the console
+	if err != nil && err.Error() == "unexpected newline" {
+		return "", nil
+	}
+
 	return result, err
 }
 
-// prompt pulls keystone auth url, domain, username and password from stdin,
-// if they are not specified initially (i.e. equal "").
-func prompt(url string, domain string, user string, project string, password string) (gophercloud.AuthOptions, error) {
+// prompt pulls missing auth parameters from stdin, if they were not specified initially (i.e. equal "")
+func prompt(options *gophercloud.AuthOptions) error {
```

Comment:
Instead of using multiple string parameters I decided to provide here a previously created AuthOption structure and use this function just to fill missing auth options in the structure